source/Calamari.Shared/Integration/Processes/CalamariVariableDictionary.cs

```diff
@@ -31,23 +32,28 @@ public CalamariVariableDictionary(string storageFilePath, string sensitiveFilePa
                 nonSensitiveVariables.GetNames().ForEach(name => Set(name, nonSensitiveVariables.GetRaw(name)));
             }
 
-            if (!string.IsNullOrEmpty(sensitiveFilePath))
+            if (sensitiveFilePaths.Any())
```

Comment:
You could have left out this block as the foreach will just be skipped on an empty collection